### ISSUE TITLE
[Feature Request] Title Change Support

### ISSUE THREAD

**Issue by digital-synapse** [kovetskiy/mark#150]
### Problem Statement
When a page name is defined with in the header metadata, this title will be used to create / update the page in confluence.
```
<!-- Title: My Article -->
```
The problem occurs if ever the title is changed. It seems that mark uses the title as the pageId. mark will assume that this is a new page and create the new page in confluence with the new title, leaving the old page orphaned. This requires some manual cleanup in confluence to delete the orphaned pages after a title change.

### Proposal:
Some internal state management is needed. mark should record an internal history log of title changes so that if a title changes mark can correctly update the page instead of creating a new page and orphaning the old one. The history log could be a simple file that is appended each time a publish occurs, or a small sqlite database. Here is an possible format for the history log.
```
[
  {
    "timestamp" : "2021-12-29 07:22:13",
    "space": "my space",
    "parents": [ "parent page" ],
    "title": "title"
  },
  {
    "timestamp" : "2021-12-30 09:06:56",
    "space": "my space",
    "parents": [ "parent page" ],
    "title": "current title"
  }  
]
```

**Comment by kovetskiy** [comment]
The problem with such solutions is that it can't be used in classic CI scenarios where the state is not shared. 

But I understand the problem you are trying to address. 

Also, did you know that mark can update a specific page if you pass an URL to a Confluence page? You can use `-l <url>` and mark will not create a new page with the new title.

Also, what do you think about just having a few titles in the same markdown file? Like as following:

```
<!-- Title: old title -->
<!-- Title: new title -->
```
and then mark will try to search for the `new title` page, if it does not exist, it will look for `old title` instead.

**Comment by digital-synapse** [comment]
Yes I see. My proposed solution sought to minimize dependencies and assumptions which is why I avoided integrating with git directly to just check the change history there. Your solution of supporting previous title metadata might be the simplest way to resolve this.

**Comment by mtsulek** [comment]
I think just adding PageId as metadata would be a very useful thing to have. If I need fixed page in the confluence page tree structure I would just use PageId instead of the title.
